lib/_http_client.js

```diff
@@ -24,6 +24,18 @@ function ClientRequest(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
+  } else if (options instanceof url.URL) {
+    options = {
+      protocol: options.protocol,
+      host: options.host,
+      hostname: options.hostname,
+      port: options.port,
+      hash: options.hash,
+      search: options.search,
+      pathname: options.pathname,
+      path: `${options.pathname}${options.search}`,
+      href: options.href
+    };
```

Comment:
I’m not sure how I feel about the code duplication here… could we make this a generic helper, possibly even hanging off the public `url` module (if that makes transitioning `url.parse → url.URL` easier)?

I would suggest adding it as the `toJSON` method for `URL` but I have looked at https://github.com/whatwg/url/issues/137 and I’m not sure what to make of that other than that my idea would probably be silly/unrealistic/… :P